internal/cloud/backend.go

```diff
@@ -210,6 +201,26 @@ func (b *Cloud) PrepareConfig(obj cty.Value) (cty.Value, tfdiags.Diagnostics) {
 	return obj, diags
 }
 
+func newWorkspacesMappingFromFields(obj cty.Value) WorkspaceMapping {
+	WorkspaceMapping := WorkspaceMapping{}
+	// To ensure that the workspaces are defined correctly in the config
+	if workspaces := obj.GetAttr("workspaces"); !workspaces.IsNull() {
+		if val := workspaces.GetAttr("name"); !val.IsNull() {
+			WorkspaceMapping.Name = val.AsString()
+		}
+		if val := workspaces.GetAttr("tags"); !val.IsNull() {
+			err := gocty.FromCtyValue(val, &WorkspaceMapping.Tags)
+			if err != nil {
+				log.Panicf("An unexpected error occurred: %s", err)
+			}
+		}
+		if val := workspaces.GetAttr("project"); !val.IsNull() && val.AsString() != "" {
+			WorkspaceMapping.Project = val.AsString()
+		}
+	}
```

Comment:
Whilst I'm aware we're moving existing code here, we have a lot of nesting `if` statements and a chance to clean some of this whilst we're here to improve readability and reduce cognitive load overall.

```suggestion
  workspaces := obj.GetAttr("workspaces")
  if workspaces.IsNull() {
      return WorkspaceMapping{}
  }
  
  nameVal := workspaces.GetAttr("name")
  if !nameVal.IsNull() {
      WorkspaceMapping.Name = nameVal.AsString()
  }
  
  tagsVal := workspaces.GetAttr("tags")
  if !tagsVal.IsNull() {
      err := gocty.FromCtyValue(tagsVal, &WorkspaceMapping.Tags)
      if err != nil {
          log.Panicf("An unexpected error occurred: %s", err)
      }
  }
  
  projectVal := workspaces.GetAttr("project")
  if !projectVal.IsNull() && projectVal.AsString() != "" {
      WorkspaceMapping.Project = projectVal.AsString()
  }
```


Maybe something like this is a bit easier to handle? Early returns and explicit steps about what each bit is handling helps the reader understand what's going on.

(**Note:** Formatting may be a little off, I wrote this in the github comment ui)

Reply:
Thanks for the suggestion James! I agree, such structure might be subjectively more human-friendly. The suggestion was applied.